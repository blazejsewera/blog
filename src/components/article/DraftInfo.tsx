import type { Component } from 'solid-js'
import { cx } from '../util/classes'
import { DraftBadge } from './DraftBadge'

export interface DraftInfoProps {
  description?: string
}

export const DraftInfo: Component<DraftInfoProps> = ({ description }) => (
  <div class={cx(['mx-16', 'mb-8', 'flex'])}>
    <DraftBadge />
    <span class={cx(['text-orange-600', 'inline', 'my-auto', 'ml-4'])}>
      {description ?? 'This is a draft article. It may be incomplete.'}
    </span>
  </div>
)
