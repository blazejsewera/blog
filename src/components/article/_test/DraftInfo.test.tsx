import { render } from '@solidjs/testing-library'
import { DraftInfo } from '../DraftInfo'

describe('<DraftInfo />', () => {
  it('renders properly without description', () => {
    const { container, unmount } = render(() => <DraftInfo />)
    expect(container).toMatchSnapshot()
    unmount()
  })

  it('renders properly with description', () => {
    const description = 'description'
    const { container, unmount } = render(() => (
      <DraftInfo description={description} />
    ))
    expect(container).toMatchSnapshot()
    unmount()
  })
})
