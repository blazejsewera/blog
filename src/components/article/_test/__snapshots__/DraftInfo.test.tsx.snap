// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<DraftInfo /> > renders properly with description 1`] = `
<div>
  <div
    class="mx-16 mb-8 flex"
  >
    <div
      class="my-auto leading-none"
    >
      <div
        class="inline-flex rounded-full border-2 w-min h-min px-2 md:px-3 py-0.5 uppercase text-2xs md:text-xs border-orange-600 text-orange-600 bg-orange-100"
      >
        <div
          class="m-auto"
        >
          draft
        </div>
      </div>
    </div>
    <span
      class="text-orange-600 inline my-auto ml-4"
    >
      description
    </span>
  </div>
</div>
`;

exports[`<DraftInfo /> > renders properly without description 1`] = `
<div>
  <div
    class="mx-16 mb-8 flex"
  >
    <div
      class="my-auto leading-none"
    >
      <div
        class="inline-flex rounded-full border-2 w-min h-min px-2 md:px-3 py-0.5 uppercase text-2xs md:text-xs border-orange-600 text-orange-600 bg-orange-100"
      >
        <div
          class="m-auto"
        >
          draft
        </div>
      </div>
    </div>
    <span
      class="text-orange-600 inline my-auto ml-4"
    >
      This is a draft article. It may be incomplete.
    </span>
  </div>
</div>
`;
